packages/web-components/src/checkbox/checkbox.ts

```diff
@@ -506,4 +539,38 @@ export class Checkbox extends BaseCheckbox {
       toggleState(this.elementInternals, next, true);
     }
   }
+
+  public setAriaChecked(value: boolean = this.checked) {
+    if (this.indeterminate) {
+      this.elementInternals.ariaChecked = 'mixed';
+      return;
+    }
+
+    this.elementInternals.ariaChecked = value ? 'true' : 'false';
+  }
+
+  constructor() {
+    super();
+    this.elementInternals.role = 'checkbox';
+  }
+
+  /**
+   * Resets the form value to its initial value when the form is reset.
+   *
+   * @internal
+   */
+  formResetCallback(): void {
+    this.indeterminate = false;
```

Comment:
Should this be reading the `indeterminate` attribute value/presence? In case the attribute is present in the author's HTML code.

Reply:
The `indeterminate` property isn't mapped to an attribute, so there's no default state.

I created this [codepen](https://codepen.io/kreitlow/pen/PovMEbP/3b142af8b7c744e6a92131cf0f89806c?editors=1010) to investigate how the `indeterminate` property is handled on form reset. Since resetting the form doesn't change the property for native checkboxes, this override for `formResetCallback()` has been removed.

Reply:
Ah cool, for some reason I thought there was an `indeterminate` attribute, should've double check. Thanks!